plugins/ComposeAuth/src/commonMain/kotlin/io/github/jan/supabase/compose/auth/LoginConfig.kt

```diff
@@ -1,14 +1,45 @@
 package io.github.jan.supabase.compose.auth
 
 
-open class LoginConfig(open val serverClientId: String)
+open class LoginConfig(open val serverClientId: String) {
+
+    /**
+     * @param key
+     * @param value
+     * json data saved to table "raw_users_meta_data"
+     */
+
+    abstract class ExtraData(open val key: String, open var value: String? = null)
+}
 
 data class GoogleLoginConfig(
     override val serverClientId: String,
     val isSupported: Boolean = true,
     val filterByAuthorizedAccounts: Boolean = false,
     val associateLinkedAccounts: Pair<String, List<String>>? = null,
     val nonce: String? = null,
-) : LoginConfig(serverClientId)
+    val extraData: List<ExtraData>? = null
+) : LoginConfig(serverClientId) {
+
+    data class ID(override val key: String) : ExtraData(key)
+    data class DisplayName(override val key: String) : ExtraData(key)
+    data class FamilyName(override val key: String) : ExtraData(key)
+    data class GivenName(override val key: String) : ExtraData(key)
+    data class ProfilePic(override val key: String) : ExtraData(key)
+}
 
-fun ComposeAuth.Config.googleNativeLogin(serverClientId: String, isSupported: Boolean = true, filterByAuthorizedAccounts: Boolean = false, associateLinkedAccounts: Pair<String, List<String>>? = null, nonce: String? = null) = GoogleLoginConfig(serverClientId, isSupported,filterByAuthorizedAccounts,associateLinkedAccounts, nonce)
+fun ComposeAuth.Config.googleNativeLogin(
+    serverClientId: String,
+    isSupported: Boolean = true,
+    filterByAuthorizedAccounts: Boolean = false,
+    associateLinkedAccounts: Pair<String, List<String>>? = null,
+    nonce: String? = null,
+    extraData: List<LoginConfig.ExtraData>? = null
```

Comment:
not sure if we really need a list, we can just use a raw JsonObject as the variable and data. The data is requested by Supabase automatically and the dev can add more to that. The [other login](https://github.com/supabase-community/supabase-kt/blob/a2ebf914d25950b9b78900f1019fab18333c39cb/GoTrue/src/commonMain/kotlin/io/github/jan/supabase/gotrue/GoTrue.kt#L175) methods also do it like that. Sorry if I said something confusing earlier